File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.py
# Last updated: 09/07/2026, 10:10:56
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        for i in range(len(haystack) - len(needle) + 1):
            for j in range(len(needle)):
                if not haystack[i+j] == needle[j]:
                    break
            else:
                return i
        return -1


        

File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.java
// Last updated: 09/07/2026, 10:10:51
class Solution {
    public int strStr(String haystack, String needle) {
        int m = haystack.length();
        int n = needle.length(); 

        for(int i = 0;i<=m-n;i++){
            String sub = haystack.substring(i,i+n);
            if(sub.equals(needle)){
                return i;
            }
        }
        return -1;
    }
}